support-frontend/assets/pages/contributions-landing/components/ContributionThankYou/ContributionThankYouHeader.jsx

```diff
@@ -39,22 +43,53 @@ const directDebitSetupText = css`
 
 type ContributionThankYouHeaderProps = {|
   name: string | null,
-  showDirectDebitMessage: boolean
+  showDirectDebitMessage: boolean,
+  paymentMethod: PaymentMethod,
+  contributionType: ContributionType,
+  amount: string,
+  currency: IsoCurrency,
+  thankyouPageHeadingTestVariant: boolean,
 |};
 
 const MAX_DISPLAY_NAME_LENGTH = 10;
 
 const ContributionThankYouHeader = ({
   name,
   showDirectDebitMessage,
+  paymentMethod,
+  contributionType,
+  amount,
+  currency,
+  thankyouPageHeadingTestVariant,
 }: ContributionThankYouHeaderProps) => {
-  const shouldShowName = name && name.length < MAX_DISPLAY_NAME_LENGTH;
+
+  const title = (): string => {
+    const nameAndTrailingSpace: string = name && name.length < MAX_DISPLAY_NAME_LENGTH ? `${name} ` : '';
+    // Do not show special header to paypal/one-off as we don't have the relevant info after the redirect
+    const payPalOneOff = paymentMethod === 'PayPal' && contributionType === 'ONE_OFF';
```

Comment:
could we remove these from the test by making the targetPage just the TY page and using `canRun` to exclude them? hmmm I can't tell if that has access to the redux state...

Reply:
unfortunately it seems we cannot rely on access to that info there. Even the function that decides which tests to put the user in doesn't have it

Reply:
I _think_ it's ok for these users to be in the test because they'll be evenly split between control/variant.
Also we can probably exclude them in the analysis if necessary

Reply:
yeah makes sense 👍